File: editorial/chap06/python/answer_B36.py
# 입력
N, K = map(int, input().split())
S = input()

# ON으로 되어 있는 것의 개수를 센다
numON = 0
for i in range(N):
	if S[i] == '1':
		numON += 1

# 답을 출력
if numON%2 == K%2:
	print("Yes")
else:
	print("No")

File: editorial/chap06/cpp/answer_B36.cpp
#include <iostream>
#include <string>
using namespace std;
 
int N, K;
string S;
 
int main() {
	// 입력
	cin >> N >> K;
	cin >> S;
 
	// ON으로 되어 있는 것의 개수를 센다
	int numON = 0;
	for (int i = 0; i < N; i++) {
		if (S[i] == '1') numON += 1;
	}
 
	// 답을 출력
	if (numON % 2 == K % 2) cout << "Yes" << endl;
	else cout << "No" << endl;
	return 0;
}
